openmls/src/group/mls_group/app_ephemeral.rs

```diff
@@ -0,0 +1,200 @@
+    /// NOTE: The main single_group_test_framework functionality can't be used in this test,
+    /// since the capabilities need to be set to include ProposalType::AppEphemeral
+    #[openmls_test]
+    fn test_app_ephemeral() {
+        let group_id = GroupId::from_slice(b"Test Group");
+
+        let alice_provider = &Provider::default();
+        let bob_provider = &Provider::default();
+
+        // Include the AppEphemeral proposal type in the LeafNode capabilities
+        let capabilities =
+            Capabilities::new(None, None, None, Some(&[ProposalType::AppEphemeral]), None);
+
+        // Define the MlsGroup configuration
+        let mls_group_create_config = MlsGroupCreateConfig::builder()
+            .ciphersuite(ciphersuite)
+            .use_ratchet_tree_extension(true)
+            // add to leaf node capabilities
+            .capabilities(capabilities.clone())
+            .build();
+
+        // Generate credentials with keys
+        let (alice_credential, alice_signer) = generate_credential(
+            b"Alice".to_vec(),
+            ciphersuite.signature_algorithm(),
+            alice_provider,
+        );
+
+        let (bob_credential, bob_signer) = generate_credential(
+            b"Bob".to_vec(),
+            ciphersuite.signature_algorithm(),
+            bob_provider,
+        );
+
+        // Generate KeyPackage for Bob with the correct LeafNode capabilities
+        let bob_key_package = KeyPackage::builder()
+            .leaf_node_capabilities(capabilities)
+            .build(ciphersuite, bob_provider, &bob_signer, bob_credential)
+            .unwrap();
+
+        // === Alice creates a group ===
+        let mut alice_group = MlsGroup::new_with_group_id(
+            alice_provider,
+            &alice_signer,
+            &mls_group_create_config,
+            group_id,
+            alice_credential.clone(),
+        )
+        .expect("An unexpected error occurred.");
+
+        // === Alice adds Bob ===
+        let welcome = match alice_group.add_members(
+            alice_provider,
+            &alice_signer,
+            &[bob_key_package.key_package().clone()],
+        ) {
+            Ok((_, welcome, _)) => welcome,
+            Err(e) => panic!("Could not add member to group: {e:?}"),
+        };
+        alice_group.merge_pending_commit(alice_provider).unwrap();
+
+        let welcome: MlsMessageIn = welcome.into();
+        let welcome = welcome
+            .into_welcome()
+            .expect("expected the message to be a welcome message");
+
+        let mut bob_group = StagedWelcome::new_from_welcome(
+            bob_provider,
+            mls_group_create_config.join_config(),
+            welcome,
+            Some(alice_group.export_ratchet_tree().into()),
+        )
+        .expect("Error creating StagedWelcome from Welcome")
+        .into_group(bob_provider)
+        .expect("Error creating group from StagedWelcome");
+
+        let message_bundle = alice_group
```

Comment:
Some comment here would be nice, since we get to the part where we actually do the proposal.